src/main/java/it/polimi/telco/controllers/ConfirmationServlet.java

```diff
@@ -0,0 +1,50 @@
+package it.polimi.telco.controllers;
+
+import it.polimi.telco.model.Subscription;
+import it.polimi.telco.model.User;
+import it.polimi.telco.services.SubscriptionService;
+import jakarta.ejb.EJB;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.annotation.WebServlet;
+import jakarta.servlet.http.HttpServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+import java.io.IOException;
+
+@WebServlet(
+        name = "confirmation",
+        urlPatterns = {"/confirmation"}
+)
+public class ConfirmationServlet extends HttpServlet {
+    @EJB(name = "it.polimi.telco.services/SubscriptionService")
+    private SubscriptionService subscriptionService;
+
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        Subscription subscription = (Subscription) request.getSession().getAttribute("subscription");
+        if (subscription == null) {
```

Comment:
we leave for now like this and I will refactor if needed

Reply:
That is the "if" I'm referring to in the above comment